JS-Algorithms/challenges.js

```diff
@@ -42,6 +51,25 @@ const COUNTRY_NAMES = ["Germany", "Norway", "Island", "Japan", "Israel"];
 
 const circularArray = (index) => {
   // YOUR CODE HERE...
+
+  let COUNTRY_COPY = [...COUNTRY_NAMES];
+  const COUNTRY_NAMES_LENGHT = COUNTRY_COPY.length;
+
+  if (index === COUNTRY_NAMES_LENGHT) return [...COUNTRY_COPY];
+
+  const GetIndex = ({ OldIndex }) =>
+    OldIndex > COUNTRY_NAMES_LENGHT
+      ? OldIndex % COUNTRY_NAMES_LENGHT
+      : OldIndex;
+
+  const IndexToReplace = GetIndex({ OldIndex: index });
+
+  const SPLICED_ITEMS = COUNTRY_COPY.slice(0, IndexToReplace);
+  const FILTERED_ITEMS = COUNTRY_COPY.filter(
+    (item) => !SPLICED_ITEMS.includes(item)
+  );
```

Comment:
Creating a copy of COUNTRY_NAMES is good if you want to avoid mutating the original array. However, since you're using slice and concat, there's no need to use COUNTRY_COPY as an intermediate variable.

here are some resources that you can review:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice